backbone.js

```diff
@@ -70,6 +70,23 @@
 
   // Proxy Underscore methods to a Backbone class' prototype using a
   // particular attribute as the data argument
+  var modelMatcher = function(attrs) {
+    var matches = _.matches(attrs);
+    return function(model) {
+      return matches(model.attributes);
+    };
+  };
+  var modelProperty = function(key) {
+    return function(model) {
+      return model == null ? void 0 : model.get(key);
+    };
+  };
+  var cb = function(iteratee) {
+    if (_.isFunction(iteratee)) return iteratee;
+    if (_.isObject(iteratee) && !(iteratee instanceof Backbone.Model)) return modelMatcher(iteratee);
```

Comment:
Actually the `collection.indexOf` test failed when trying to locate a Backbone.Model in the collection. Since it's an Object, it was catched by the second `if` and was converted to a function. In order to keep the indexOf behaviour unchanged, I added the `!instanceof` test.